File: 2022/04/py/2.py
import re
data = open("i.txt").read().split("\n")

n = 0
for l in data:
    m = re.search(r"(\d*)-(\d*),(\d*)-(\d*)", l)
    a, b, c, d = [int(n) for n in m.groups()]
    if (a<=c and b>=d) or (a>=c and b<=d) or a<=c<=b or a<=d<=b or c<=a<=d or c<=b<=d:
        n+=1
print(n)

File: 2022/04/js/2.js
const fs = require("fs");
let data = fs.readFileSync("i.txt").toString().split(/\r?\n/);
console.log(data.reduce((acc, l) => {
    [a, b, c, d] = l.split(/,|-/).map(c => parseInt(c, 10));
    if ((a <= c && c <= b) || (a <= d && d <= b) || (c <= a && a <= d) || (c <= b && b <= d)) return acc + 1;
    return acc;
}, 0));